sdks/go/container/tools/buffered_logging.go

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tools
+
+import (
+	"context"
+	"os"
+	"strings"
+)
+
+const INITIAL_LOG_SIZE int = 255
+
+// BufferedLogger is a wrapper around the FnAPI logging client meant to be used
+// in place of stdout and stderr in bootloader subprocesses. Not intended for
+// Beam end users.
+type BufferedLogger struct {
+	logger  *Logger
+	builder *strings.Builder
```

Comment:
Consider not using a pointer here, and just having the value. Avoids needing to explicitly initialize it, since the zero value is safe.

Reply:
Yeah I was back and forth on this but my Rust with go is showing. I'll change it and probably take out the constructor function since we wouldn't be doing anything extra in it